front_end/core/react_native/RNReliabilityMetrics.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import {type ReactNativeChromeDevToolsEvent} from './generated/type_defs/ReactNativeChromeDevToolsEventTypes.js';
+
+type RNReliabilityMetrics = {
+  sendEvent: (event: ReactNativeChromeDevToolsEvent) => void,
+};
+
+export const RNReliabilityMetrics = ((): RNReliabilityMetrics => {
+  function sendEvent(_event: ReactNativeChromeDevToolsEvent): void {
+  }
+
+  return {
+    sendEvent,
+  };
+})();
```

Comment:
Why not

```suggestion
export const RNReliabilityMetrics: RNReliabilityMetrics = {
  sendEvent(_event: ReactNativeChromeDevToolsEvent): void {}
};
```

?

Also, reusing the exact same name for a type and value seems problematic.

Reply:
Also: How are we going to actually inject a `sendEvent` handler here?

Reply:
This plan was for this to be a pub/sub API, so sending the actual events won't be here:

https://github.com/EdmondChuiHW/rn-chrome-devtools-frontend2/commit/ecf72b6e95405e423f50e3c88d7e65f173c2beff#diff-c62722ca9646ac9ec9e205faf447976c33eccbea2ce1a39791bf6f95d5041233R17

The closure style is my muscle memory for private variables + singleton; I will change it to a class + instance export instead as JS has real private fields now.

Will rework this PR to focus only on what are Reliability Metrics, how are safe defaults included, and the actual listener wiring (similar to https://github.com/facebook/react/pull/22276)